File: Common/Entities/Session.cs
using System;
using System.Collections.Generic;

namespace openspace.Common.Entities
{
    public class Session
    {
        public bool AttendanceEnabled { get; set; } = true;

        public string CreatedAt { get; set; } = DateTime.Now.ToShortDateString();

        public bool FreeForAll { get; set; }

        public int Id { get; set; }

        public string Name { get; set; }

        public ICollection<Room> Rooms { get; set; } = new List<Room>();

        public ICollection<Slot> Slots { get; set; } = new List<Slot>();

        public bool TeamsAnnouncementsEnabled { get; set; } = false;

        public ICollection<Topic> Topics { get; set; } = new List<Topic>();

        public bool VotingEnabled { get; set; }

        public ICollection<FeedbackCategory> FeedbackCategories { get; set; } = new List<FeedbackCategory>();
    }
}

File: src/Website/src/app/models/session.ts
import { Room } from "./room";
import { Slot } from "./slot";
import { Topic } from "./topic";

export class Session {
  public id: number;
  public name: string;
  public topics: Topic[];
  public rooms: Room[];
  public slots: Slot[];
  public createdAt: string;
  public freeForAll: boolean;
  public votingEnabled: boolean;
  public attendanceEnabled: boolean;
  public teamsAnnouncementsEnabled: boolean;

  constructor() {
    this.topics = [];
    this.slots = [];
    this.rooms = [];
  }
}
